### DISCUSSION TITLE
Any success linking tabulator to Google firestore database?

### DISCUSSION THREAD

**Discussion by NicoN13** [anvilistas/tabulator#96] [Q&A]
Hi all,

Has anyone had any success linking the anvil tabulator to Google Firestore? I'd like to have it load data lazily in the same manner that data can be loaded from the anvil datatables. Not sure where to start, and couldn't find any suggestions in the tabulator official documentation.

**Comment by lewysigns** [comment]
How do you currently access your firestore database?

I ask because the format of how you retrieve you data determines how you have to link up with tabulator AND I am ignorant to firestore.


The official [documentation ](https://tabulator.info/docs/5.4/data#import) supports: JSON,CSV, and Arrays.
You can also use http [endpoints ](https://tabulator.info/docs/5.4/data#ajax-methods) to pull data into your table. If you need a custom importer, you can also write your own as well.
